### PR TITLE
Adding Muon Rochester Correction v3 for Run2

### PR DESCRIPTION
**Pull request by sroychow**
Reference twiki:-
https://twiki.cern.ch/twiki/bin/view/CMS/RochcorMuon

Rochester Muon corrections for Run2 version3. The module saves 3 branches in the output
Corrected Pt, Variations UP and down of the corrected pt.

### CODE REVIEW

File: python/postprocessing/modules/common/muonScaleResProducer.py [hunk 1]
```diff
@@ -37,54 +36,54 @@ def endJob(self):
...
-                    roccor.kScaleAndSmearMC,
-                    mu.charge, mu.pt, mu.eta, mu.phi, mu.nTrackerLayers, u1, u2
-                ) for mu in muons)
-            if self.is_2017:
-                pt_err = list(
-                    mu.pt * mk_safe(
-                        roccor.kScaleAndSmearMCerror,
-                        mu.charge, mu.pt, mu.eta, mu.phi, mu.nTrackerLayers, u1, u2
-                    ) for mu in muons)
+            pt_corr=[]
```
Comment (arizzi) [inline python/postprocessing/modules/common/muonScaleResProducer.py]: why is "u1" reused for all muons? is that in the prescription? isn't this smearing all muons in an event in a correlated way?

File: python/postprocessing/modules/common/lepSFProducer_v2.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
+import ROOT
```
Comment (arizzi) [inline python/postprocessing/modules/common/lepSFProducer_v2.py]: isn't lepSFProducer_v2 factorized out in another PR? are there cross dependencies witht he rocc?

File: python/postprocessing/modules/common/muonScaleResProducer.py (hunk 1 above)
Comment (sroychow) [reply python/postprocessing/modules/common/muonScaleResProducer.py]: Yes you are correct. The u1 random number has to different for all the muons. I will close this PR. Some updates are required.

File: python/postprocessing/modules/common/lepSFProducer_v2.py (hunk 2 above)
Comment (sroychow) [reply python/postprocessing/modules/common/lepSFProducer_v2.py]: There should not be any dependencies. I have not made the PR for lepSFProducer_v2.

File: python/postprocessing/modules/common/lepSFProducer_v2.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        elif(len(self.histos) > 1):
+            sf_lep_stat = [ self.getSFstaterr(lep.pt, lep.eta) for lep in leptons ]
+            sf_lep_syst = [ 0.005 for lep in leptons ]
+        else:
+            sf_lep_stat = [ 0.005 for lep in leptons ]
+            sf_lep_syst = [ 0.005 for lep in leptons ]
+
+        sf_lep_statUp   = list(map(add, sf_lep, sf_lep_stat)) 
+        sf_lep_statDown = list(map(sub, sf_lep, sf_lep_stat))
+        sf_lep_systUp   = list(map(sub, sf_lep, sf_lep_syst))
```
Comment (arizzi) [inline python/postprocessing/modules/common/lepSFProducer_v2.py]: sub -> add

**Pull request closed without merging**